### DISCUSSION TITLE
Add Java programmes to Application Launcher

### DISCUSSION THREAD

**Discussion by ldesousa** [regolith-linux/regolith-desktop#1078] [Q&A] [answered]
Hi there,

I install Java programmes by creating a corresponding folder in `/opt` and then creating a symbolic link to the main executable in `/usr/bin`. E.g.:

`sudo ln -s /opt/archi/Archi /usr/bin/Archi`

In i3 this is enough for a programme to be available to the launcher. However that is not the case with Regolith, none of the Java programmes I have installed appear in the Application Launcher, I must start them from the command line. How can it be done?

Thank you.

**Comment by kgilmer** [comment, marked as answer]
Hi @ldesousa ,  ilia has two launching modes: Apps (<super> space) and CLI (<super><shift> space) programs.  I think anything on `/usr/bin` should show up in CLI programs.  For something to be launchable from the Apps mode, the app needs to have a corresponding `.desktop` file in a specific user directory: `~/.local/share/applications/`.  This page contains more details: https://wiki.archlinux.org/title/Desktop_entries

**Reply by ldesousa** [reply]
Laborious, but works, thanks. How can I access the CLI mode in Ilia?
